network/proxy/mucp/mucp.go

```diff
@@ -79,155 +84,126 @@ func readLoop(r server.Request, s client.Stream) error {
 	}
 }
 
-func (p *Proxy) getRoute(service string) ([]string, error) {
-	// converts routes to just addresses
-	toNodes := func(routes []table.Route) []string {
-		var nodes []string
-		for _, node := range routes {
-			address := node.Address
-			if len(node.Gateway) > 0 {
-				address = node.Gateway
-			}
-			nodes = append(nodes, address)
+// toNodes returns a list of node addresses from given routes
+func toNodes(routes map[uint64]table.Route) []string {
+	var nodes []string
+	for _, node := range routes {
+		address := node.Address
+		if len(node.Gateway) > 0 {
+			address = node.Gateway
 		}
-		return nodes
+		nodes = append(nodes, address)
 	}
+	return nodes
+}
 
+func (p *Proxy) getRoute(service string) ([]string, error) {
 	// lookup the route cache first
 	p.RLock()
 	routes, ok := p.Routes[service]
-	// got it!
 	if ok {
-		p.RUnlock()
```

Comment:
I think removing this will now cause it to hold read locks since it's supposed to return here

Reply:
Not sure what you mean here -- the `Unlock()` has moved below the routes submap on line 108.

Reply:
Oh, I got it now. Noticed that as soon as I started modifying code. Thanks!